src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -1323,24 +1398,39 @@ RelationalDBInterfaceSqliteImp::getTransaction(
     }
 
     /* else use shard databases */
-    std::variant<AccountTx, TxSearched> res(TxSearched::unknown);
-    iterateTransactionBack(
-        {}, [&](soci::session& session, std::uint32_t index) {
-            std::optional<ClosedInterval<uint32_t>> range1;
-            if (range)
-            {
-                uint32_t low = std::max(range->lower(), firstLedgerSeq(index));
-                uint32_t high = std::min(range->upper(), lastLedgerSeq(index));
-                if (low <= high)
-                    range1 = ClosedInterval<uint32_t>(low, high);
-            }
-            res = ripple::getTransaction(session, app_, id, range1, ec);
-            /* finish iterations if transaction found or error detected */
-            return res.index() == 1 &&
-                std::get<TxSearched>(res) != TxSearched::unknown;
-        });
+    if (app_.getShardStore())
+    {
+        std::variant<AccountTx, TxSearched> res(TxSearched::unknown);
+        auto db = txMetaDb_->checkoutDb();
 
-    return res;
+        if (auto const shardIndex =
+                ripple::getShardIndexforTransaction(*db, id);
+            shardIndex)
```

Comment:
shardindex being checked twice. Can be..
```if (auto const shardIndex = ripple::getShardIndexforLedger(*db, ledgerHash))```

Reply:
Done